rasa/core/actions/action.py

```diff
@@ -591,7 +591,103 @@ def action_response_format_spec() -> Dict[Text, Any]:
                     "type": "array",
                     "items": {
                         "type": "object",
-                        "properties": {"event": {"type": "string"}},
+                        "properties": {"event": {"type": "string"}, "timestamp": {}},
+                        "oneOf": [
+                            {
+                                "properties": {
+                                    "event": {"const": "user"},
+                                    "text": {"type": "string"},
+                                    "input_channel": {},
+                                    "message_id": {"type": "string"},
+                                    "parse_data": {
+                                        "type": "object",
+                                        "properties": {
+                                            "text": {"type": "string"},
+                                            "intent": {
+                                                "type": "object",
+                                                "properties": {
+                                                    "name": {"type": "string"},
+                                                    "confidence": {"type": "number"},
+                                                },
+                                            },
+                                            "entities": {
+                                                "type": "array",
+                                                "items": {
+                                                    "type": "object",
+                                                    "properties": {
+                                                        "start": {"type": "integer"},
+                                                        "end": {"type": "integer"},
+                                                        "name": {"type": "string"},
+                                                        "confidence": {
+                                                            "type": "number"
+                                                        },
+                                                        "extractor": {"type": "string"},
+                                                        "value": {"type": "object"},
+                                                    },
+                                                },
+                                            },
+                                        },
+                                    },
+                                }
+                            },
+                            {
+                                "properties": {
+                                    "event": {"const": "action"},
```

Comment:
`metadata`, `timestamp`, `hide_rule_turn` and `action_text` are missing

Reply:
I'll add `action_text` and `hide_rule_turn`, however I thought that if `metadata` and `timestamp` are mentioned in L594 `properties`, I don't need to add them in 'oneOf', was that incorrect?